specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/applicationGateway.json

```diff
@@ -1895,6 +1920,28 @@
       ],
       "description": "Allows to disable rules within a rule group or an entire rule group."
     },
+    "ApplicationGatewayFirewallExclusion": {
+      "properties": {
+        "matchVariable": {
+          "type": "string",
+          "description": "The variable to be excluded."
+        },
+        "selectorMatchOperator": {
+          "type": "string",
```

Comment:
How far down in the stack are these strings vs. enums? If they were enums here, I think the auto generated API's will have built in validation and auto-completion, so we should consider the tradeoffs of enums vs strings here.

Reply:
we have string defination on the nrp and also have validation there, so don't want to make the transfer in the nrp code. It will be too long for us.